### ISSUE TITLE
New lightweight tutorial for newcomers (based on Killercoda)

### ISSUE THREAD

**Issue by oleg-nenashev** [keptn/tutorials#205]
There are multiple tutorials available on the Keptn website but all of them are quite complex and require a lot of dependencies. In particular these tutorials do not quite work on Windows or on MacBook M1. They're hard to follow and they don't help newcomer users to quickly understand what is the project about and how to get the value out of it.

I suggest creating a fully new tutorial that would focus on onboarding newcomers within 10 minutes or so. Some properties:

- The tutorial uses a minimum set of technologies and fully open source: Keptn, Prometheus
- The tutorial demonstrates a very simple but realistic use case for Keptn usage. The proposal is to have auto-remediationof a single container application based on Prometheus metrics
- ideally the tutorial should be portable between other platforms and runnable on all common user hardware where they want to evaluate the project. It is unlikely to be the target hardware

After this new tutorial is created, the existing tutorials can be refactored to learning path soo that each of them shows particular feature or integration.

## Scope

- [ ] - Update the tutorials to reference quickstart guide: https://github.com/keptn/tutorials/pull/209
- [ ] - Automate the environment setup in the quichkstart

**Comment by agardnerIT** [comment]
Thanks to work from @StackScribe and @afzal442 we now have: https://killercoda.com/keptn/scenario/keptn-end-to-end-delivery

## The Tutorial

- Runs on killercoda (cloud infrastructure) so nothing to install
- Infra and tutorial are all written "as code"
- Installs on a ubuntu base image so we are in control of the k8s version used
- Uses minimal services (no helm-service which removes the dependency on Istio, no jmeter-service)
- Uses job-executor service to run **Helm** and **k6**
- Demonstrates multistage delivery with approval steps, preproduction quality gates, release validation quality evals & self healing (helm update deployment with scaling pods)

## Outstanding Concerns / Unknowns
- Nowhere could I find a concurrent user limit for the free tier
- Nowhere could I see information on "we have this much capacity available for free tier"

## Alternatives
- Do not use SaaS hosting - stay with "localhost methodology" asking users to install on their infrastructure
- ~Katacoda~ (no longer exists)
- [AppSembler](https://www.appsembler.com/virtual-labs)
- [Instruqt](https://instruqt.com)
- Keptn created infra (Keptn creates the backend infra and offers a web terminal like [ttyd](https://github.com/tsl0922/ttyd) or [xterm.js](https://xtermjs.org)

## Advantages of This Approach
- Killercoda team have been very helpful and responsive with my questions on Slack
- Mutual exposure: Killercoda receives exposure from a large OSS project Keptn and Keptn can be listed on Killercoda homepage as a "supported tutorial"
- Eliminates user setup error (insufficient CPU / RAM on lower power laptops)
- Tutorial as code eliminates differences between environments
- Everything in browser - no additional tooling installs required
- Automated tear-down
- Easy to clone and create dev environments for upgrades

## Downsides / Concerns of This Approach

- [By their own admission](https://killercoda.com/faq), Killercoda is "so early in Beta stage that we could also call it Alpha!"
- Killercoda environments only last for 1hr max. or 20mins of keyboard activity (whichever comes first)
- Requires users to log in to Killercoda (via GitHub, Gitlab, Gmail or Email)
- Compute is limited to 4GB machines in free tiers
- Time limited scenarios
- Must link to Killercoda website (cannot embed tutorials into keptn.sh)
- No native statistics offered

Some of these limitations disappear if Killercoda offer higher level memberships for OSS projects OR Keptn pays for "Professional" tier:

- Embedding is possible. No longer required to login
- Larger base VMs available
- Statistics available
